### PR TITLE
Add MList

### LINKED ISSUE #22: Add Lists
- Material Design Guideline
  - [Lists](https://material.io/design/components/lists.html)

### PR DESCRIPTION
**Pull request by flyofsky**
### Description of Change ###
This PR is for adding MList

### Issues Resolved ### 
- fixes #22

### API Changes ###
Added
 - MList

### Platforms Affected ### 
- Tizen

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
 - single line
Selection_016

 - double line
Selection_017

 - triple line
Selection_018

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/Tizen.NET.MaterialComponents/Components/MList.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
+    public enum MListStyle
+    {
+        OneLine,
+        DoubleLine,
+        TripleLine
+    }
+
+    public class MItem
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MList.cs]: Create MItem.cs for `MItem` class.

File: src/Tizen.NET.MaterialComponents/Components/MList.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+        public string Title { get; set; }
+
+        public string SubText { get; set; }
+
+        public string MetaText { get; set; }
+
+        public EvasObject Control { get; set; }
+
+        public MItem(string title, MListStyle style = MListStyle.OneLine)
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MList.cs]: Any reason to provide optional parameter only for style?

File: src/Tizen.NET.MaterialComponents/Components/MList.cs (hunk 1 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MList.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MList.cs (hunk 2 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MList.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MItem.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        public string Title { get; set; }
+
+        public string SubText { get; set; }
+
+        public string MetaText { get; set; }
+
+        public EvasObject Control { get; set; }
+
+        public MItem(string title, string icon = null)
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MItem.cs]: avoid optional parameter as a public API. Is is necessary>

File: src/Tizen.NET.MaterialComponents/Components/MList.cs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            get => _backgroundColor;
+            set
+            {
+                _backgroundColor = value;
+                Color effectiveColor = _backgroundColor.IsDefault ? _defaultBackgroundColor : _backgroundColor;
+                base.BackgroundColor = effectiveColor;
+            }
+        }
+
+        public IList<MItem> Items
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MList.cs]: what about using `ObservableCollection<MItem>` instead?

File: src/Tizen.NET.MaterialComponents/Components/MItem.cs (hunk 3 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MItem.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MList.cs (hunk 4 above)
Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MList.cs]: Done

File: src/Tizen.NET.MaterialComponents/Components/MList.cs [hunk 5]
```diff
@@ -0,0 +1,184 @@
+using ElmSharp;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.IO;
+
+namespace Tizen.NET.MaterialComponents
+{
+    public class MList : GenList, IColorSchemeComponent
```
Comment (rookiejava) [inline src/Tizen.NET.MaterialComponents/Components/MList.cs:8]: As `MList` inherit `GenList`, all public APIs of `GenList` can be exposed. For example, user can use `Append()` to add item. How do you think about this?

Comment (flyofsky) [reply src/Tizen.NET.MaterialComponents/Components/MList.cs:8]: 'MList' thinks it is to provide a 'Material Design', not to constrain its function.
So I think it's a class that makes the 'GenList' look like a 'Material Design', not just a 'Material Design' list.

**Comment by shyunMin** [comment]
New PR(#52) has been opened according to design changes

**Pull request closed without merging**